ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -644,14 +643,17 @@ validateCandidate lgrDB tracer cfg invalidPoints
       fitCandidateSuffixOn curChain candSuffix
 
     -- | Make a list of all the points after from the given point (inclusive)
-    -- in the candidate fragment.
+    -- in the candidate fragment and turn it into a map from hash to slot that
+    -- can be added to 'cdbInvalid'.
     --
     -- PRECONDITON: the given point is on the candidate fragment.
-    pointsStartingFrom :: HasCallStack => Point blk -> [Point blk]
-    pointsStartingFrom pt = case AF.splitBeforePoint suffix pt of
+    hashesStartingFrom :: HasCallStack
+                       => Point blk -> Map (HeaderHash blk) SlotNo
+    hashesStartingFrom pt = case AF.splitBeforePoint suffix pt of
         Nothing                  -> error "point not on fragment"
-        Just (_, startingFromPt) ->
-          map headerPoint $ AF.toOldestFirst startingFromPt
+        Just (_, startingFromPt) -> Map.fromList $
+          map (\hdr -> (blockHash hdr, blockSlot hdr)) $
```

Comment:
Otherwise with a lazy `Map` then the elements here will retain the whole `hdr` and not just the slot number!